tests/rptest/e2e_tests/flink_scale_test.py

```diff
@@ -232,11 +431,31 @@ def get_commit_requests_counts(method, nodes) -> list[MetricSamples]:
                          f"{json.dumps(metric_per_node, indent=2)}")
         self.logger.info(f"Total: {total_commit_requests}")
 
-        # Number of total transactions should be not less than
-        # transaction_count / total_events. I.e. >0.1 in this case.
-        # I.e. flink should be able to send at least 1 transaction per 1 ms
-        assert 0.1 < total_commit_requests / target_total_events, \
+        # With default parallelization optimizations currently set
+        # transaction rate from single node should be at least 1 per checkpoint
+        # mostly due to the fact that we starting workloads in order.
+        # Ideally, ratio should be >= vcpu count. I.e. each vcpu should
+        # generate at least 1 transaction per checkpoint. But since there is
+        # a swarm of topics involved, they will fight for cpu time,
+        # so 1 transaction per node is reasonable.
+
+        # Multiple nodes involved (5), so expected ratio is
+        # 1 transaction per each node per per 1 ms interval.
+        # Also, we account for 20% losses on the network
+        rate_per_node = 1
+        target_transaction_rate = rate_per_node * total_nodes * 0.8
+        self.logger.info(
+            f"Calculated target transaction rate: {target_transaction_rate} per 1 ms"
+        )
+        # Actual rate is simple total msg divided by transactions
+        actual_transaction_rate = topics_total_events / total_commit_requests
+        self.logger.info(
+            f"Actual transaction rate: {actual_transaction_rate} per 1 ms")
+        assert target_transaction_rate < actual_transaction_rate, \
             "Flink failed to generate at least 1 transaction " \
-            "per checkpoint each 1 ms"
+            f"per half of the task managers ({rate_per_node}) " \
+            f"at each of the nodes ({total_nodes})" \
+            "per checkpoint (1 ms): " \
+            f"{target_transaction_rate} < {actual_transaction_rate}"
```

Comment:
This code is same as above, factor it out into a helper?

Reply:
Yes, will do.